### PR TITLE
Calendar based versioning

### PR DESCRIPTION
**Pull request by MariaSemple**
### Intent

Addresses [Pro #2652](https://github.com/rstudio/rstudio-pro/issues/2652)

### Approach

* Add a notice to the About Page that builds are Daily/Preview/Release w/ notice about supportability of dailies & previews.
* Add `daily` or `preview` to the package build name.
* Add a separate upload bucket for dailies and previews.
* Set the major version to 2021-07 and the minor version to 0.
* Add a text file (lack of blank line at the end intentional) that controls the build phase (BUILDTYPE) of the branch.

### Automated Tests

This will impact all of our automated testing, but I didn't add any for the new Daily/Preview tagging in the about page. If we have an integration test for that page, we could check for it on dailies/previews.

### QA Notes

* Preview & release builds are separate from daily builds now, and need to be downloaded from the internal S3 bucket (for the time being, better solution to come)
* Release packages should _not_ have release in the name, only dailies and previews.

### Checklist

- [x] If this PR adds a new feature, or fixes a bug in a previously released version, it includes an entry in `NEWS.md` 
- [x] If this PR adds or changes UI, the updated UI meets [accessibility standards](https://github.com/rstudio/rstudio/wiki/Accessibility)
- [x] A reviewer is assigned to this PR (if unsure who to assign, check Area Owners list)
- [x] This PR passes all local unit tests

### CODE REVIEW

File: src/gwt/src/org/rstudio/studio/client/application/ui/AboutDialogContents.java [hunk 1]
```diff
@@ -77,6 +77,17 @@ public AboutDialogContents(ProductInfo info, ProductEditionInfo editionInfo)
       productName.setText(editionInfo.editionName());
       copyrightYearLabel.setText("2009-" + info.copyright_year);
 
+      // Warn that dailies and previews aren't supported
+      if (info.build_type.equals("Daily") || info.build_type.equals("Preview")) {
```
Comment (kevinushey) [inline src/gwt/src/org/rstudio/studio/client/application/ui/AboutDialogContents.java:81]: I wonder if this should check `build_type != "Release"`? Assuming that will be the only "official" label used for releases.

File: Jenkinsfile [hunk 2]
```diff
@@ -76,8 +76,18 @@ def s3_upload(type, flavor, os, arch) {
...
   packageFile = renamedFile
 
+  def buildType = sh (
+    script: "cat BUILDTYPE",
+    returnStdout: true
+  ).trim().toLowerCase()
+
+  def buildDest =  "s3://rstudio-ide-build/${flavor}/${os}/${arch}/"
+  if (buildType != "daily") {
+    buildDest =  "s3://rstudio-ide-build-internal/${buildType}/${flavor}/${os}/${arch}/"
```
Comment (kevinushey) [inline Jenkinsfile:86]: Nit: we put this into a bucket called "internal", but I assume these will still be user-visible (just from elsewhere)?

File: Jenkinsfile [hunk 3]
```diff
@@ -6,8 +6,8 @@ properties([
                               artifactNumToKeepStr: '',
                               daysToKeepStr: '',
                               numToKeepStr: '100')),
-    parameters([string(name: 'RSTUDIO_VERSION_MAJOR', defaultValue: '1', description: 'RStudio Major Version'),
-                string(name: 'RSTUDIO_VERSION_MINOR', defaultValue: '5', description: 'RStudio Minor Version'),
+    parameters([string(name: 'RSTUDIO_VERSION_MAJOR', defaultValue: '2021-07', description: 'RStudio Major Version'),
```
Comment (kevinushey) [inline Jenkinsfile:9]: One worry I have here is that some existing code might be assuming that the major version component is a parsable number; this change will break that assumption.

Is having the year as the major version, and month as the minor version, a suitable compromise?

**Review by kevinushey — COMMENTED** [review]
Overall LGTM. One other question: what is the build type we might want to give to one-off builds that patch a specific customer issue? Are those still considered as "release" builds, or should we consider an alternate name?

File: Jenkinsfile (hunk 2 above)
Comment (MariaSemple) [reply Jenkinsfile:86]: These will only become user visible when we upload them as we do with previews and releases

File: Jenkinsfile (hunk 3 above)
Comment (MariaSemple) [reply Jenkinsfile:9]: I don't believe the scheme for the version is debatable. We must have the minor version able to increment for patches (0, 1, 2, etc.). This is a change across all RStudio products, AFAIK

Comment (kevinushey) [reply Jenkinsfile:9]: Why not use the patch version component for patches?

Comment (MariaSemple) [reply Jenkinsfile:9]: We don't actually have a patch version, that just ends up being build number. I also don't think we can change the versioning scheme. The major version needs to be YYYY-MM

Comment (kevinushey) [reply Jenkinsfile:9]: Gotcha. The main thing that I think needs to be validated then is that `RStudio.Version()` still produces a parsable version in the `version` field, and that `rstudioapi::getVersion()` also produces a parsable version (but I think that just reads that same field).

Comment (MariaSemple) [reply Jenkinsfile:9]: Got it, I will test this

Comment (MariaSemple) [reply Jenkinsfile:9]: The output:
```R
> RStudio.Version()
$mode
[1] "server"

$version
[1] ‘2021.7.0.1011’

$release_name
[1] "Ghost Orchid"

> rstudioapi::getVersion()
[1] ‘2021.7.0.1011’
```

File: dependencies/linux/install-dependencies-focal [hunk 4]
```diff
@@ -73,8 +73,8 @@ sudo apt-get -y install \
 # Java 8 (not in official repo for focal)
 sudo add-apt-repository -y ppa:openjdk-r/ppa
 sudo apt-get update
-sudo apt-get -y install openjdk-8-jdk
-sudo update-alternatives --set java /usr/lib/jvm/java-8-openjdk-amd64/jre/bin/java
+sudo apt-get -y install openjdk-11-jdk
```
Comment (jmcphers) [inline dependencies/linux/install-dependencies-focal:76]: Does not seem related to the intent -- was this included intentionally?

**Review by jmcphers — APPROVED** [review]
This LGTM but merging it will require us to make changes in a couple places so that builds still flow correctly. In particular:

- S3 buckets need to get created, and seed versions need to be added, for the version bump script (otherwise these won't be versioned)
- The dailies build page needs to be adjusted to pull these new builds (currently it only pulls "1.5" builds)

File: dependencies/linux/install-dependencies-focal (hunk 4 above)
Comment (MariaSemple) [reply dependencies/linux/install-dependencies-focal:76]: Yes, I had to change this to get the ant build working. Would you prefer I exclude it?
